api/src/test/java/org/openmrs/api/handler/PatientDataUnvoidHandlerTest.java

```diff
@@ -144,4 +149,39 @@ public void handle_shouldNotUnvoidTheOrdersAndEncountersThatNeverGotVoidedWithTh
 		Assert.assertNotNull(testOrder.getVoidReason());
 		
 	}
+	
+	/**
+	 * @see PatientDataUnvoidHandler#handle(Patient,User,Date,String)
+	 */
+	@Test
+	@Verifies(value = "should unvoid the members associated with the patient", method = "handle(Patient,"
+			+ "User,Date,String)")
+	public void handle_shouldUnvoidMembersAssociatedWithThePatient() throws Exception {
+		executeDataSet(COHORT_XML);
+		CohortService cs = Context.getCohortService();
+		Cohort cohort = cs.getCohort(2);
+		Patient patient = Context.getPatientService().getPatient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		cs.addMembershipToCohort(cohort, membership);
+		Assert.assertTrue(cohort.contains(patient));
+		
+		patient = Context.getPatientService().voidPatient(patient, "Void Reason");
+		Assert.assertTrue(patient.getVoided());
+		
+		Assert.assertTrue(membership.getVoided());
+		Assert.assertNotNull(membership.getDateVoided());
+		Assert.assertNotNull(membership.getVoidedBy());
+		Assert.assertNotNull(membership.getVoidReason());
+		
+		//TODO: unvoiding logic needs to be addressed
+		//Context.getPatientService().unvoidPatient(patient);
```

Comment:
I wanted to clear a doubt with @wluyima. Basically to address [this comment](https://github.com/openmrs/openmrs-core/commit/0b662c88e8352da41bb3ac6525557b4801bf3e48#commitcomment-21178110) that I added in the earlier commit. I will remove it.